src/librustc/mir/transform/dataflow.rs

```diff
@@ -168,12 +134,14 @@ impl<F: Lattice> ::std::ops::IndexMut<BasicBlock> for Facts<F> {
 }
 
 /// Analyse and rewrite using dataflow in the forward direction
-pub fn ar_forward<'tcx, T, P>(cfg: &CFG<'tcx>, fs: Facts<P::Lattice>, mut queue: BitVector)
--> (CFG<'tcx>, Facts<P::Lattice>)
-// FIXME: shouldn’t need that T generic.
-where T: Transfer<'tcx, P::Lattice, TerminatorOut=Vec<P::Lattice>>,
-      P: DataflowPass<'tcx, Transfer=T>
+pub fn ar_forward<'tcx, T, R>(cfg: &CFG<'tcx>, fs: Facts<T::Lattice>, transfer: T, rewrite: R)
+-> (CFG<'tcx>, Facts<T::Lattice>)
+where T: Transfer<'tcx>,
+      R: Rewrite<'tcx, T::Lattice>
 {
+    let mut queue = BitVector::new(cfg.len());
+    queue.insert(START_BLOCK.index());
```

Comment:
Similarly here, I do not believe that beginning at the start block is always correct/desired/etc.